hedera-node/src/main/java/com/hedera/services/txns/crypto/CryptoAdjustAllowanceTransitionLogic.java

```diff
@@ -133,16 +135,23 @@ private ResponseCodeEnum validate(TransactionBody cryptoAllowanceTxn) {
 	 *
 	 * @param cryptoAllowances
 	 * 		newly given crypto allowances in the operation
-	 * @param ownerAccount
-	 * 		owner account
+	 * @param payerAccount
+	 * 		account of the payer for this adjustAllowance txn
 	 */
-	private void adjustCryptoAllowances(final List<CryptoAllowance> cryptoAllowances, final Account ownerAccount) {
+	private void adjustCryptoAllowances(final List<CryptoAllowance> cryptoAllowances, final Account payerAccount) {
 		if (cryptoAllowances.isEmpty()) {
 			return;
 		}
-		final Map<EntityNum, Long> cryptoMap = new TreeMap<>(ownerAccount.getCryptoAllowances());
+
 		for (final var allowance : cryptoAllowances) {
+			final var owner = allowance.getOwner();
+
+			final var accountToAdjust = fetchOwnerAccount(owner, payerAccount, accountStore, entitiesChanged);
+			final Map<EntityNum, Long> cryptoMap = new TreeMap<>(accountToAdjust.getCryptoAllowances());
```

Comment:
Combined with the fact that `Account.setCryptoAllowances()` [also creates a new `TreeMap`](https://github.com/hashgraph/hedera-services/blob/de9315c53ac6fbe63fd736c31c43ab393f21843b/hedera-node/src/main/java/com/hedera/services/store/models/Account.java#L326), we are creating up to 2 unnecessary `TreeMap`'s per `allowance` processed for a given `owner`.

Instead we can just have a `Account#getMutableCryptoAllowances()` method that returns a `TreeMap` (creating one if the current map is `emptyMap()`); and then here, directly update that same map for each crypto `allowance` being adjusted.

Reply:
Good point. Will get this done.